### DISCUSSION TITLE
Clarification on `cosmic-config` versioning

### DISCUSSION THREAD

**Discussion by HeitorAugustoLN** [pop-os/libcosmic#725] [Q&A] [answered]
I'm working on a software project to manage COSMIC desktop/app configurations via the command line, that is [cosmic-ctl](https://github.com/HeitorAugustoLN/cosmic-ctl), a tool similar KDE's `kreadconfig` and `kwriteconfig`.

One key aspect I'd like to know is how the configuration version in `cosmic-config` is updated. Should it be tied to:

1. Breaking changes in the configuration schema?
2. The version of the software?
3. Some other criteria?

This clarification will help me design my tool for COSMIC software. Additionally, I plan to create an application using `libcosmic` for the COSMIC desktop in the future, so understanding the versioning strategy will ensure my work is consistent with the broader COSMIC ecosystem.

**Comment by mmstick** [comment, marked as answer]
Unless there is a breaking change in one of the config files, then incrementing the version isn't necessary. Applications will only read the versioned config files that they were programmed to at compile-time.

**Reply by HeitorAugustoLN** [reply]
Thanks for the quick answer and clarification! :)
